### DISCUSSION TITLE
qna2024/

### DISCUSSION THREAD

**Discussion by giscus** [cechallenge/cechallenge.github.io#3] [Q&A]
# qna/

Samsung CE Challenge site

https://cechallenge.github.io/qna/

**Comment by poult-lab** [comment]
Hello Sponsor,

Thank you for the opportunity to have this challenge. I believe it is very significant. Could you please provide the English version of the challenge as well? I appreciate it.

Thank you.

**Reply by cechallenge** [reply]
We have completed the English version of our website.

**Comment by KuriosKat** [comment]
I applied without needing an orin when I applied, but I wonder if I can rent an orin because it is different from the schedule.(The team name is MoCvD)

**Reply by cechallenge** [reply]
Please reapply as we previously informed you via email.

**Comment by bluebear78** [comment]
Thanks for hosting a great contest, but the rules are too strict and weird.
No tuning, no model architecture modification, no pruning, no quantization.
Is there any room for these strict rules to be relaxed?

**Reply by cechallenge** [reply]
A. The purpose of this Challenge is to compare how effectively you perform the same workload.
I'm sorry for this, but we don't allow you to change the model that is workload.

**Comment by KuriosKat** [comment]
Okay sir.

2024년 7월 31일 (수) 오후 2:49, cechallenge ***@***.***>님이 작성:

**Comment by delibae** [comment]
@cechallenge 평가 방식의 불일치: 현재 대회의 test_script.py 및 test_dataset.jsonl 파일은 **텍스트 생성 작업을 통한 벤치마크를 상정**하고 있는 것으로 보입니다. 그러나 open dataset 으로 지정된 OpenBookQA와 같은 일부 데이터셋의 평가 코드를 살펴보면, **텍스트 생성이 아닌 각 선택지의 로짓(logit) 값을 통해 정확도를 측정하는 방식**을 사용하고 있습니다.
```python
# OpenBookQA dataset evaludation code
#https://github.com/allenai/OpenBookQA/blob/b51971646e9371a61508d9953fc706645e194a71/obqa/models/qa/multi_choice/qa_multi_choice_esim.py#L114
        self._accuracy = CategoricalAccuracy()

...
            self._accuracy(label_logits, label)

```
- 벤치마크 데이터셋의 비일관성: 19개의 벤치마크 데이터셋이 통일된 규격을 갖추고 있지 않습니다. 일부는 텍스트 생성 작업을 통한 text output 비교 방식을, 다른 일부는 로짓 값 비교 방식을 사용합니다. 예를 들어:
    - 로짓 값 비교: HellaSwag, OpenBookQA 등
    - 텍스트 생성: AGI Eval 등

- 만약 데이터셋 처리 코드까지 최종 제출물에 포함된다면 다음과 같은 내용이 필요할 것 같습니다.(**만약 그렇지 않다면 정확한 제출물 규격을 알려주시면 감사하겠습니다.**)
    -  a) 데이터셋 규격 통일: 정확한 평가를 위해 모든 데이터셋의 규격을 통일할 필요가 있습니다. 
    - b) 평가 방식 명확화: 각 데이터셋별 정확한 평가 방식을 명시해야 합니다. 
    - c) Closed 및 Open 데이터셋 일관성: Docker 이미지를 통한 정량적 평가를 위해 closed 데이터셋과 open 데이터셋의 규격을 일치시켜야 합니다.

관련 내용 검토 부탁드립니다. 감사합니다.
(jetson agx orin 대여를 위한 계획서 작성에 영향이 있을 것 같아  빠른 답변 부탁드립니다 🤗)

**Reply by cechallenge** [reply]
**데이터셋 처리 코드는 실행 속도 평가 부분에 포함되지 않습니다.**
시간 측정에 포함되는 부분은 데이터 로드와 실행(inference) 부분이 포함됩니다. 예를 들어, test_script.sh 내 data_load 와 pipe가 이에 해당됩니다.
리더보드 평가 이후 본 평가에서는 accuracy 측정 및 이를 위한 inference 후 데이터셋 처리는 주최측에서 진행할 예정입니다.
(test_script.py에는 리더보드 운영을 위해 간단한 accuracy 측정이 포함되어 있습니다.)

참가자 분들은 데이터 로드와 실행 부분에 대한 최적화를 진행하셔야 합니다. 이 때, 반드시 Transformer의 pipeline을 사용할 필요는 없습니다. **단, pipe 실행 결과와 동일한 form의 output을 생성하는 것만 허용됩니다.**
만약 해당 부분을 다른 최적화한 package를 사용한 경우에는 최종 제출물에 실행 코드도 포함되어야 합니다.

위 내용은 Rules 에 업데이트 하도록 하겠습니다.

**Reply by delibae** [reply]
제가 알기로는 text-generation pipeline 의 경우에는 HellaSwag 등과 같은 logit 값을 비교하여 정확도를 측정하는 태스크를 처리하지 못합니다. 즉 phi-3-medium  report에 나와있는 accuracy 와 동일한 결과물을 뽑아내지 못하는 것으로 알고있습니다. 만약 huggingface pipeline 과 같은 form  의 output (text output or token indices) 을 내는 것만 허용이 되는 것이라면 19개의 benchmark dataset 전체가 아닌 평가에 사용하는 데이터셋에 대한 명확한 재정의가 필요할 것 같습니다.

**Reply by cechallenge** [reply]
text-generation은 예시를 위해 설정된 값으로, dataset에 따라 변경될 수 있습니다. (question-answering 등)
dataset별로 나오는 pipe 함수의 output form이 유지되어야 한다고 이해하시면 될 것 같습니다.

Accuracy에 대해 말씀해주신 부분은 내부적으로 추가 논의 후에 정확히 답변 드리도록 하겠습니다.

**Reply by delibae** [reply]
정확도 측정관련된 부분은 text-generation pipeline 만 사용했을 때 기준으로 말씀드린거여서 question-answering 같은 것도 사용가능하면 accuracy 측정이 가능할 것 같습니다. 친절한 답변 갑사드립니다 :)

**Comment by cechallenge** [comment]
Q. No tuning, no model architecture modification, no pruning, no quantization.
Is there any room for these strict rules to be relaxed?

**Reply by cechallenge** [reply]
A. The purpose of this Challenge is to compare how effectively you perform the same workload.
I'm sorry for this, but we don't allow you to change the model that is workload.

**Comment by matt1398** [comment]
@cechallenge 

안녕하세요, 업데이트된 규칙 및 데이터셋 확인 후 질의사항이 생겨 문의 드립니다. huggingface pipeline 을 사용하게 되고 accuracy test code 를 주최측에서 관리한다면 batch size 에 대한 구체적인 명시가 필요할 것 같습니다. 만약 batch size 가 1, 즉 순차적으로 처리하는 경우가 아니라면, 데이터셋 별로 context length 별 차이가 생겨 메모리 관련된 문제가 생길 수 있고, 이 문제는 주최측에서 accuracy test code 를 관리하기에 참여자 입장에서는 통제가 어려울 수 있을 것 같습니다.

관련해서 구체적인 규칙에 대한 답변 주시면 감사하겠습니다! 😃

**Reply by cechallenge** [reply]
Batch size에 대한 명시는 없습니다. 참여자들이 자유롭게 설정 가능한 수치입니다. 본 문제의 출제 의도 상 메모리 관련된 문제는 해결해야 하는 요소 중 일부분일 수 있습니다. Accuracy는 batch_size에 관계없이 측정 가능한 요소입니다.

**Comment by widiba03304** [comment]
@cechallenge

안녕하세요. 프레임워크 관련 문의 드립니다.

대회에서 규정하고 있는 프레임워크는 "최적화를 위해 사용 할 딥러닝 프레임워크는 PyTorch 2.x입니다."라고 적혀있는데, 
- PyTorch C++/cuda extension을 사용하는 것
- torchscript등 torch model을 compile해서 사용하는 것
- libtorch를 사용하는 것
등 범위도 인정이 되는 것인지 궁금합니다.

감사합니다.

**Reply by cechallenge** [reply]
Pytorch 를 기반으로 하는 extension 및 라이브러리 사용은 허용합니다.
다만, 모델을 컴파일을 통하여 다른 형태로 변환하는 것은 모델 변환에 해당하여 불가합니다.

**Reply by widiba03304** [reply]
@cechallenge 답변 감사드립니다.
아래에서 tensorrt 사용은 가능하다고 답변해주셨는데, tensorrt도 compilation을 하는 것으로 알고 있습니다. 
[https://github.com/pytorch/TensorRT?tab=readme-ov-file#quickstart](https://github.com/pytorch/TensorRT?tab=readme-ov-file#quickstart)

torchscript, onnx, tensorrt 등 compilation은 weight와 activation에 차이를 거의 주지 않는 것으로 아는데 모델 변환에 해당하는 기준이 어떤 것인지 궁금합니다.

**Reply by cechallenge** [reply]
모델 변환은 양자화, pruning등의 모델 최적화 뿐만이 아니라 graph 변환 등을 통하여 기존의 라이브러리를 사용하는 단순한 최적화를 허용하지 않는 것입니다.

**Reply by widiba03304** [reply]
@cechallenge 죄송하지만, 아직 정확히 이해가 가지 않아서 다시 질문 드립니다.
1. Graph 변환은 어떤 것을 말하는 것인지요? 예를 들어, FlashAttention을 사용하는 경우에는 여러 레이어들을 하나의 레이어로 바꾸어서 계산해야할텐데 (kernel fusion), 이 경우는 graph 변환인가요?
2. 만약 Graph 변환이 torch.nn.Module에서 다른 것으로의 변환이라면, TensorRT는 가능하고 torchscript는 불가능한 이유가 무엇인지요? 같은 graph-level compilation이 있고 기존의 라이브러리를 사용하는 것 아닌가요? 또한 torch-tensorrt를 사용하면 이미 torchscript로 변환 과정이 존재하게 됩니다. (https://github.com/pytorch/TensorRT/blob/577c5c443bc58d52a70ce2005e35e5f697644eb2/py/torch_tensorrt/_compile.py#L202C1-L206C46)
3. 만약 사용가능한 TensorRT가 torch-tensorrt(https://github.com/pytorch/TensorRT/tree/577c5c443bc58d52a70ce2005e35e5f697644eb2) 가 아닌 tensorrt(https://github.com/NVIDIA/TensorRT) 인 경우 이는 사용 프레임워크가 torch 2.x가 아니어서 룰에 위반이 되는 것으로 보입니다.
4. TensorRT의 사용이 libnvinfer.so를 의미하는 것이고 torch c++ extension을 사용하면 문제가 없어보이는데, 이를 의도하신 것일까요?

Graph라고 하는 것이 워낙에 모호한 개념이어서 혼동이 있는 듯 합니다.

**Reply by cechallenge** [reply]
내부 논의 결과, torchscript를 허용하기로 결정했습니다. 
또한 TensorRT는 Torch-TensorRT 버전으로만 사용 가능합니다.

**Reply by widiba03304** [reply]
@cechallenge 그러면 FlashAttention과 같은 layer fusion 또한 가능한 것인지요? 정확하게는 PyTorch 프레임워크 내에서 weight, activation을 바꾸지 않는 선에서 최적화(i.e., floating point operation으로 생기는 numerical errors만 허용한)는 모두 가능한 것일까요?

**Reply by cechallenge** [reply]
네. 맞습니다.  
FlashAttention과 같이 이론적으로 연산 결과가 동일하다면, 모두 가능합니다.

**Comment by JWHer** [comment]
안녕하세요. 대회 규정 관련해 질문드립니다. @cechallenge

대회 규칙에 Framework 항목을 보면,
"최적화를 위해 사용 할 딥러닝 프레임워크는 PyTorch 2.x입니다." 이라고 적혀 있는데 TensorRT를 활용하는 것은 불가능한가요?

감사합니다.

**Reply by cechallenge** [reply]
TensorRT 사용은 가능합니다.

**Reply by bluebear78** [reply]
TensorRT내에서 디폴트로 quantization 단계가 있는걸로 아는데, tensorRT로 경량화할때, 양자화 기능을 사용하면 룰 위반인가요?

**Reply by cechallenge** [reply]
네, 어떠한 quantization도 허용되지 않습니다.

**Comment by bluebear78** [comment]
안녕하세요

참가신청을 했는데, 메일이나 메세지가 안와서 문의드립니다.
신청이 제대로 됐다면 연락이 따로 오나요? 아니면 제대로 참가신청이 안된걸까요?

**Reply by cechallenge** [reply]
확인해보니 참가신청이 안 되었습니다. 다시 확인하셔서 참가 신청 부탁 드립니다.

**Comment by EvelynKimm** [comment]
안녕하세요.

Rules에서 기기대여 관련 안내가 취소선으로 표시되었음을 확인했습니다. 혹시 대여 가능한 기기의 수량이 모두 소진된 것인가요?

답변 부탁드립니다. 감사합니다.

**Reply by cechallenge** [reply]
네. 준비된 수량은 모두 소진되었습니다.

**Comment by juseongjin** [comment]
안녕하세요.

보유한 기기가 NVIDIA Jetson AGX Orin 64GB인데, 32GB가 아니라면 사용 불가능할까요?

**Reply by cechallenge** [reply]
사용 가능합니다만 메모리나 클럭 등 스펙을 규정된 기기와 비슷하게 제한할 것을 권장합니다.
최종 평가는 저희가 보유한 NVIDIA Jetson AGX Orin 32GB로 이루어집니다.

**Comment by LimHyeongSoo** [comment]
안녕하세요.

참가 신청을 하였는데 아무런 메일이나 연락이 오지 않아 문의를 드립니다!
혹시 참가 신청을 하게 되면 연락이 따로 오게 되나요? 팀명은 NDRW로 하였습니다

**Reply by cechallenge** [reply]
메일 드렸습니다. 다시한번 메일함 확인 부탁드립니다.

**Comment by juseongjin** [comment]
혹시 참가신청 확인 가능할까요? 팀명은 MISys 입니다.

**Reply by cechallenge** [reply]
참가 접수되었습니다.

**Comment by widiba03304** [comment]
안녕하세요. 혹시 모든 operation들이 full precision (FP32)로 진행이 되어야하는지 궁금합니다.
예를 들어, TF32, FP16, BF16 등 integer가 아닌 다른 precision을 사용하는 것은 괜찮을까요?

- https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html?highlight=wmma#element-types-and-matrix-sizes
- https://docs.nvidia.com/cuda/ampere-tuning-guide/index.html#improved-tensor-core-operations

위 두 문서를 보면 Tegra architecture가 가지고 있는 Ampere tensor core의 경우 fp32는 accumulation precision으로만 존재하는 것으로 보여 cuda core만 사용하는 것을 의도하신 것인지 궁금합니다.

**Reply by widiba03304** [reply]
제가 헷갈렸습니다. huggingface 기본 config는 bfloat16으로 되어있는 것을 확인했습니다.

**Comment by csehydrogen** [comment]
안녕하세요,
transformers의 pipeline은 매우 많은 task(text-generation 등) 및 설정을 지원하고 있는데요.
(https://huggingface.co/docs/transformers.js/en/api/utils/generation#module_utils/generation.GenerationConfig)
pipeline을 직접 구현해서 사용하는 경우 모든 task와 모든 설정을 지원하도록 구현해야 하는지 궁금합니다.
지원해야 하는 task와 설정값의 범위가 정해져 있다면 (예를 들어, do_sample은 항상 False라든지) 명시해주시면 좋을 것 같습니다.
감사합니다.

**Reply by cechallenge** [reply]
지원해야 하는 task는 공지해드린 30개 중에서 임의로 선택합니다.
성능(accuracy)을 판별해야 하므로 대다수의 task에 대하여 do_sample은 false로 진행할 것입니다.
실제 테스트에서 사용할 task는 규정 상 알려드리지 못하는 점 양해 바랍니다.

**Comment by yspkm** [comment]
안녕하세요. 

혹시 나중에 평가하실 때의 평가 환경 관련하여서 문의 드립니다.
1. 팀마다 평가 수행시에 clean boot로 수행하는지 여부 (혹은 drop caches를 하며 테스트를 수행하는지)
2. power mode (ex. MAXN) 또는 jetson_clocks 적용 여부
3. desktop gui 화면이 떠 있는 상태인지 여부 (service gdm stop을 하고 평가를 수행하는지)

그 외에 아래 정보도 공유해주실 수 있는지 문의 드립니다.
1. IDLE 상태에서의 상주 메모리 크기 정보를 공유해주실 수 있는지
2. baseline에서의 수행 시간 정보를 공유해주실 수 있는지

감사합니다.

**Reply by cechallenge** [reply]
각 팀 마다 Mode 0(=MAXN) 하 동일한 환경에서 평가를 수행합니다. 
GUI는 disable 상태에서 진행 예정입니다만, 사용하시는 환경에 대하여 필요하다고 생각하는 세팅은 제출하시는 문서에 명기해주시면 좋습니다.

그 외 문의 주신 정보는 공유가 불가함을 말씀드립니다만, 환경은 Jetson 초기화를 한 상태에서 바로 평가를 수행합니다.

**Reply by yspkm** [reply]
안녕하세요.

답변 감사드립니다.

추가로 확인하고 싶은 사항이 있어 다시 문의 드립니다. 혹시 jetson_clocks 적용 여부에 대해서는 답변을 주시지 않아, 이 부분에 대해서도 확인 가능하실지 여쭙고 싶습니다. 

감사합니다.

**Reply by cechallenge** [reply]
Mode 0(MAXN) 환경에서 수행하며, jetson_clock 스크립트는 사용하지 않을 예정입니다. 
참고 부탁드립니다.

**Comment by gajagajago** [comment]
안녕하세요. 
사용 가능한 framework를 pytorch 2.x로 한정한다는 것이 아래 두 케이스를 모두 제한하는 것인가요?
1) model file을 변환하는 framework (e.g., llama.cpp) 사용하는것
2) torch c/c++로 핵심 로직을 작성하는 것 (i.e., pytorch는 wrapper로만 사용)

**Reply by cechallenge** [reply]
1. 모델 변환은 torchscript 만 현재 허용합니다.
2. 가능합니다. 다만, 연산 결과의 차이가 크게 나지 않아야 합니다

**Comment by park-jongwon** [comment]
안녕하세요,
Speculative Decoding의 허용 여부에 대해 다시 질문드리려고 합니다.

8월 초에 메일로 Speculative Decoding은 가능하며, 곧 Rules에도 가능하다고 명시 예정이라는 답변을 받았었는데,
아직 Rules에는 관련 내용이 없는 것 같습니다.

1. Speculative Decoding을 허용하는 것이 맞는지,
2. 허용한다면 draft model의 학습도 전면 허용되는 것인지 질문드리고 싶습니다.

**Reply by cechallenge** [reply]
1. 허용합니다.
2. 만약 Speculative decoding 자체를 구현한다면, draft model은 무엇을 사용해도 상관없지만 검증 모델은 본 과제에서 제시한 모델을 사용하여야 하며, 연산 결과 성능(accuracy)이 허용 범위 내어야 합니다.

**Comment by NayaKim** [comment]
안녕하세요?
hugging face의 모델을 받아서, 동일한 bfloat16 type으로 변환한 gguf를 사용하는 것은 가능한가요?
weight에 변화가 있지 않고 메타데이터를 저장하는 파일 구조가 바뀌는 것 뿐이라 연산 결과에도 차이가 없습니다.
적절한 출처를 제공한다면 오픈소스 라이브러리의 사용이 가능하다고 했는데, llama.cpp 같은 응용이 사용하는 gguf 형식의 파일을 사용해서 decoding을 진행할 수 있는지 질문 드립니다.

**Reply by cechallenge** [reply]
데이터 포멧 변경은 허용됩니다. 
하지만, torch 기반의 프레임워크를 활용하셔야 됩니다.

**Comment by kevin99-debug** [comment]
안녕하세요, 혹시 참가신청 확인 가능할까요?
팀명 정동주로 접수했습니다

**Reply by kevin99-debug** [reply]
13일에 접수했는데 아무 안내 메일도 받지못해서 문의드립니다

**Reply by cechallenge** [reply]
안녕하세요,
정상적으로 접수되어 있습니다. 
KIT 대여가 없는 경우, 별도 안내 메일은 발송하지 않고 있습니다. 
감사합니다.

**Comment by JWHer** [comment]
@cechallenge
안녕하세요 대회 일정과 제출물 관련해서 질문드립니다.
Timeline에 보면 대회 종료일이 9월 13일, 코드 및 PPT 제출이 9월 24일로 되어있습니다. 각각 Leaderboard 제출물, 최종 제출물을 내는 것이 맞나요?

즉, 13일 대회 종료 이후에는 문서 작업만 가능하고 성능 개선 작업은 불가능한지 질문드립니다.

**Reply by cechallenge** [reply]
안녕하세요 

참가자분들께 문제 해결을 위한 충분한 시간을 드리고자 대회 최종 마감일을 9월 27일(금)로 연장하였습니다.
공지되었으니 확인 부탁드립니다. 

대회에 관심 가져주시고 참여해주셔서 감사드립니다.

**Comment by Songhyeonjin0507** [comment]
안녕하세요. 모델 사용 홈페이지에 접속을 시도했습니다.
회원님의 계정이 잠겼습니다. 계정을 잠금 해제하려면 지원 담당자에게 문의한 다음 다시 시도하세요.
라고 뜨면서 접속이 되지 않습니다.

**Reply by cechallenge** [reply]
안녕하세요.
로그인 없이 홈페이지 접속 가능합니다.
혹시, Phi-3에 대한 자세한 설명이 필요하시면 아래 주소를 참고해주시고, 모델은 HuggingFace를 통해서 다운로드 받으실 수 있습니다.
https://github.com/microsoft/Phi-3CookBook

**Comment by YangDongHeon** [comment]
안녕하세요. 대회 제출 형식 및 평가 방식에 대해 문의드립니다.

대회 Rules에 보면 적절한 출처 제공시 오픈소스 코드와 라이브러리 사용을 허용한다고 나와있습니다. 이러한 라이브러리의 경우 실험 환경에서 실행하기 위해서는 적절한 설치 및 빌드 과정등이 있어야 코드를 돌릴 수 있을 것 같은데요, 그렇다면 제출물 형태가 어떤 식으로 되는지 궁금합니다. Docker image인가요? 아니면 빌드된 파일을 포함한 폴더 자체를 제출하나요?

한편, NVIDIA JetPack6.0 문서를 확인하면 CUDA 12.2.1, cuDNN 8.9.4로 명시되어 있는데요, 실험환경에서의 추가적인 빌드의 과정을 허용한다면 PyTorch2.x제약을 지키면서 다른 CUDA버전을 사용할 수도 있을 것 같은데, 이러한 것이 허용이 되는지도 궁금합니다.

**Reply by cechallenge** [reply]
안녕하세요. 
제출 방식은 docker image 입니다.
CUDA와 cuDNN의 경우, jetPACK6 버전인 12.2.1과 8.9.4만 가능합니다.

**Comment by hayoungim** [comment]
안녕하세요. triton 써도 되나요?

**Reply by cechallenge** [reply]
안녕하세요.
triton 사용 가능합니다. 
감사합니다.

**Comment by macto94** [comment]
안녕하세요. 
최적화의 타겟이 "데이터 로드"와 "inference"에 해당하는 부분이라고 공지가 되어있습니다.
최종 검증시, 데이터 로드 부분이 다음 방식 중 어떤 방식으로 동작하는지 알 수 있을까요?
- datasets hub에서 다운로드해 local에 캐시해오는 방식
- 현재 공개된 'test_dataset.jsonl'을 사용하는 것 처럼 local에 저장된 파일을 load하는 방식
  - 해당 방식으로 동작한다면, jsonl이 아닌 다른 format을 사용해도 되는지
  - 만약 다른 format이 허용된다면, jsonl의 데이터를 다른 format으로 변환하는 시간도 최종 total latency에 포함이 되는지
위의 사항들이 궁금합니다. 감사합니다.

**Reply by cechallenge** [reply]
안녕하세요
최종 검증 방식은 내부 방침에 따라 공개가 불가합니다. 데이터셋은 jsonl format만을 사용해주세요.

**Comment by richbang** [comment]
안녕하십니까,
출력 결과에 대해서 질문이 있습니다.

1. JSON 형식으로만 출력 되면 괜찮은 것인가요?
Key값에 generated_text가 아니라 다른 형식의 값이 와도 상관 없다는 것이죠?

2. [{'아무 키나 상관 없음 ': "모델이 응답한 내용(True, False 혹은 생성된 자연어 ex) To solve the equation 2x + 3 = 7 ... "}]

3. dataset마다 모델의 출력 형식이 다를 수 있음을 감안할 때(key값 등..), 임의로 모델의 출력 내용을 parsing 해서 json의 key-value로 재가공해도 되나요?

4. 데이터셋의 shot의 수는 모델의 벤치마크 테이블을 참고하여 똑같이 적용해서 시간을 측정하나요?

**Reply by cechallenge** [reply]
안녕하세요.
출력 포맷은 예시 코드의 pipe실행 결과와 동일해야 합니다.
pipeline 생성할 때 입력된 task형태에 맞는 리턴 형태에 맞춰서 만들어 주시면 문제없습니다.
즉, text-generation의 경우 generated_text가 key가 되어야 합니다.

**Comment by easyhardhoon** [comment]
안녕하세요.
대회 규칙에 "Jetson AGX Orin kit에 외장 메모리를 추가하여 사용 가능합니다"라고 명시되어 있는데,  외장 메모리를 사용해 swap memory를 생성하고 이를 inference 시에 활용할 수 있는지 궁금합니다. 

만약 허용되지 않는다면, 리눅스 시스템에서 가상 메모리 공간을 확장하기 위해 (emmc로부터) 자동으로 설정하는 swap memory의 크기를 조절하여 inference 시에 사용이 가능한지도 궁금합니다.

감사합니다.

**Reply by cechallenge** [reply]
안녕하세요.
Swap memory 사용 가능합니다.
추후 평가를 위해 코드, 동작 환경, 문서 제출 시 설정 방법 및 환경에 대해 자세히 명시해 주세요.

감사합니다.

**Comment by jeongjun-kim** [comment]
안녕하세요.

젯슨 리셋을 한 상태에서 실험을 진행하신다고 하셨는데 젯슨 리셋을 하면서 설치할 수 있는 패키지의 종류 유무에 따라 젯슨 리셋 후의 초기 storage 사용량이 달라지기 때문에 reset 직후 swap space를 off하였을 때 기준 jetson storage의 여유 용량이 궁금합니다. 

감사합니다.

**Reply by cechallenge** [reply]
안녕하세요. 
약 40GB 정도의 storage 여유 공간이 남아있습니다. 

감사합니다.

**Comment by seonjunn** [comment]
안녕하세요,
input prompt 및 generated output의 길이 관련해서 질문이 있습니다.

1. input에 있어서 각 task별로 어떤 system prompt가 사용될지 문의드립니다. system prompt(public dataset에서는 "Read the question and answer the following ...")를 어떻게 덧붙이냐에 따라 input길이가 달라지기 때문에 multiple choice가 아닌 task에 대해 어떤 system prompt가 사용되는지의 정보가 필요합니다.

2. output에 있어서 제공된 test_script.py에 max_new_tokens가 500개로 쓰여 있는데 해당 값이 이후 evaluation시에도 사용되는 값인지 문의드립니다. 

input prompt 또는 generated output의 길이가 길어지면 경우에 따라 single item을 처리하는 과정에서도 OOM이 올 수 있고,
팀마다 적용한 optimization approach에 따라 runtime memory footprint가 다르기 때문에 private evaluation시 OOM이 나지 않는 적절한 data를 고르는 것도 쉽지 않을 것으로 보입니다.
만약 임의의 input 길이와 임의의 output 길이에 대해 OOM 없이 inference가 support되는 구현을 필요로 한다면 그건 inference throughput을 높이는 대회 취지와 다른 방향의 engineering이 많이 들어가야 할 것 같은데, 그런 의도가 맞는지 궁금해서 문의드리게 되었습니다.

감사합니다.

**Reply by cechallenge** [reply]
안녕하세요. 

1. maximum token 길이에 대해서는 메일로 공지드리도록 하겠습니다. 
2. evaluation 에서도 사용되는 값입니다. 

감사합니다.

**Comment by yezinri** [comment]
안녕하세요,
마지막 단계에 Torch-TensorRT 적용해도 되는지 질문드립니다. ( 물론 quantization 포함입니다! )

감사합니다.

**Reply by cechallenge** [reply]
안녕하세요. 

Torch-TensorRT는 적용 가능하나, quantization 사용은 불가합니다. 

감사합니다.

**Comment by seongho-git** [comment]
안녕하세요, 
메일 통해서 전달해주신 내용 중 max token length 1K로 제한한다는 것은 output이 아닌 input만 제한한다는 말씀으로 이해했는데 혹시 맞을까요?

감사합니다.

**Reply by cechallenge** [reply]
안녕하세요.
max token length는 input length + output length를 의미합니다. 참고 바랍니다.
감사합니다.

**Comment by bsm2002** [comment]
안녕하세요,
bf16 type대신 fp16으로의 type 변환은 가능한지 질문드립니다.

감사합니다

**Reply by cechallenge** [reply]
안녕하세요. 
type 변환은 불가합니다. 

감사합니다.

**Comment by yezinri** [comment]
안녕하세요.

weight은 이미 다운 받아져 있는건가요? 

감사합니다.

**Reply by cechallenge** [reply]
안녕하세요 

모델은 이미 받아져 있습니다. 

감사합니다.

**Comment by YangDongHeon** [comment]
안녕하세요. 8월 19일 질문, 8월 20일 답변하신 QnA에서 

"데이터 포멧 변경은 허용됩니다.
하지만, torch 기반의 프레임워크를 활용하셔야 됩니다."

다음과 같이 답변을 하셨는데, 이는 데이터의 type변환이 가능하다는 뜻 아닌가요?

최근 올라온 질문에 type 변환이 불가하다고 하셔서 다시 여쭤봅니다.

감사합니다.

**Reply by cechallenge** [reply]
안녕하세요. 정확히는 gguf와 같은 모델의 포맷 변경은 허용한다로 이해하시면 될 것 같습니다. 
이 때, 데이터의 타입은 변경되어서는 안 됩니다.

**Comment by seongho-git** [comment]


**Reply by cechallenge** [reply]
안녕하세요. 

Y-phi 팀의 김성호님 맞으실까요?
Y-phi 팀은 현재 리더보드에서 확인하실 수 있으십니다.

**Comment by suho-yoo** [comment]
안녕하세요.
Blackwell 팀으로 참가 신청 했고,
Dataset평가 3차 제출 메일로 드렸는데 반영이 안 되어서
혹시 참가 신청이 되지 않은 것인지 아니면 메일 전달에 착오가 있었는지 확인 가능할까요??

**Reply by cechallenge** [reply]
리더 보드 업데이트 완료했습니다.
감사합니다.

**Comment by jeongjun-kim** [comment]
안녕하세요 titan 팀명으로 3차 제출물로 제출하였으나 리더보드의 결과로 반영되지않아 문의드립니다.
만약 누락된 것이라면 3차결과를 리더보드에 반영해주시면 감사하겠습니다.

**Reply by cechallenge** [reply]
리더 보드 업데이트 완료했습니다. 
감사합니다.

**Comment by kmjeong98** [comment]
안녕하세요.
data를 로드한 뒤 처리 순서를 변경해도 될까요? data를 sort하고 싶습니다.
감사합니다.

**Reply by cechallenge** [reply]
처리 순서는 변경 가능합니다. 하지만 변경 시간도 평가 시간 측정에 포함됩니다.

**Comment by easyhardhoon** [comment]
안녕하세요. 아래 사항들에 대해 질문 드립니다.

1. 테스트 데이터셋의 아이템 (데이터) 개수는 비공개인가요? 
2. 테스트 데이터셋은 task 별로 독립적으로 존재하나요?
3. 테스트 데이터셋의 아이템 수와 각 아이템의 토큰 수는 모든 팀들에 대해 동일하게 적용되나요?

감사합니다.

**Reply by cechallenge** [reply]
1. 데이터의 개수는 비공개 입니다.
2. Task는 여러 개 존재하지만 데이터셋은 한 묶음으로 취급합니다.
3. 모두 동일하게 같은 데이터셋을 적용합니다.

**Comment by bluebear78** [comment]
안녕하세요. 

데이터 전처리 -> 데이터 로드 -> 모델 형태로 처리하려고 한다면, 전치리 과정도 처리시간에 포함인가요?

감사합니다.

**Reply by cechallenge** [reply]
네. 전처리 과정도 처리시간에 포함됩니다.

**Comment by seongho-git** [comment]
안녕하세요,
Leaderboard에서 사용된 데이터셋과 최종 심사에서 사용될 데이터셋은 동일한지 궁금합니다.

감사드립니다.

**Reply by cechallenge** [reply]
모두 동일한 데이터 셋을 사용합니다. 
감사합니다.

**Comment by junsung3737** [comment]
안녕하세요,
Leader board와 대회 결과의 연관성에 대해 질문이 있어 올립니다.
혹시 중간 제출로 leader board에 올라간 결과가 있는 상태에서, 최종 대회 제출한 결과가 오동작이라면 그 전 leader board 결과를 인정받을 수 있는지 궁금합니다. 인정여부에 따라 최종 결과 제출의 형태가 달라질 수 있을 것 같아 질문남깁니다.
감사합니다.

**Reply by cechallenge** [reply]
중간 제출 결과는 인정되지 않습니다. 
최종 결과만 인정됩니다. 
감사합니다.

**Comment by bluebear78** [comment]
안녕하세요.

19개 데이터셋을 보니 특수 문자 및 데이터 케이스가 많은 것 같아서 문의드립니다.

json file을 읽을때, encoding 방식은 utf-8, errors는 ignore로 두고 읽으면 문제없이 읽히나요?

감사합니다

**Reply by cechallenge** [reply]
안녕하세요.
비공개 데이터셋 load 관련하여 함수의 작동 방식에 대해서는 답변드리기 어렵습니다.
제공해드린 스크립트의 load_dataset 함수를 참고바랍니다.

**Comment by jiyong-j** [comment]
안녕하세요, `문제 해결을 위해 사용한 최적화 방법과 알고리즘을 설명하는 문서`에 문의 사항이 있습니다.
팀원 중 외국인이 있어서 이 문서를 **영어**로 작성하는 것을 고려 중인데, 혹시 이를 영어로 작성해도 괜찮을지 궁금합니다.

감사합니다.

**Reply by cechallenge** [reply]
영어로 작성하셔도 무관합니다. 
감사합니다.

**Comment by richbang** [comment]
모든 관계자 분들 및 수상자 분들 수고하셨습니다.
두 가지 질문이 있습니다.
최종 성적 공개 및 수상 팀 코드 공개할 예정이 있나요? 감사합니다.
수상 팀들의 소스 코드를 연구하며 공부하고 싶습니다.

**Reply by cechallenge** [reply]
안녕하세요 

참가팀의 성적은 개별 메일로 안내드릴 예정입니다. 
수상팀의 소스코드는 해당팀이 소유권을 가지고 있어 저희가 공개를 결정할 수 있는 부분이 아님을 양해 부탁드립니다. 

감사합니다.

**Comment by YangDongHeon** [comment]
안녕하세요

참가팀의 성적이 메일로 안내가 진행되었나요? 아무런 메일을 받지 못해 질문 드립니다.

감사합니다.

**Reply by cechallenge** [reply]
안녕하세요

10월 25일에 팀장(ke*******@***.**.**)님에게 송부드렸습니다.  
재전송이 필요하시면 cechallenge@samsung.com로 연락 부탁드립니다. 

감사합니다.
